### PR TITLE
[16.0][MIG] account_invoice_pricelist_sale

### PR DESCRIPTION
**Pull request by nguyenminhchien**
Depend on:
* https://github.com/OCA/account-invoicing/pull/1337

### CODE REVIEW

**Comment by StephaneMangin** [comment]
For the PR dependency, please take in account this comment https://github.com/OCA/edi/pull/724#issuecomment-1461765822

**Comment by SoniaViciana** [comment]
LGTM

File: account_invoice_pricelist_sale/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _prepare_invoice(self):
+        """Pricelist_id is set on invoice."""
+        self.ensure_one()
+        val = super(SaleOrder, self)._prepare_invoice()
```
Comment (RodrigoBM) [inline account_invoice_pricelist_sale/models/sale_order.py]: Apply PEP 3135 (28-Apr-2007) to "super()."
```suggestion
        val = super()._prepare_invoice()
```

File: account_invoice_pricelist_sale/tests/test_module.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Copyright (C) 2019 - Today: GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests.common import TransactionCase
+
+
+class TestModule(TransactionCase):
+    def setUp(self):
+        super(TestModule, self).setUp()
```
Comment (RodrigoBM) [inline account_invoice_pricelist_sale/tests/test_module.py]: Apply PEP 3135 (28-Apr-2007) to "super()."
```suggestion
        super().setUp()
```

**Review by RodrigoBM — CHANGES_REQUESTED** [review]

**Comment by RodrigoBM** [comment]
rebase please @nguyenminhchien

**Review by SoniaViciana — APPROVED** [review]

File: account_invoice_pricelist_sale/tests/test_module.py (hunk 2 above)
Comment (nguyenminhchien) [reply account_invoice_pricelist_sale/tests/test_module.py]: updated

File: account_invoice_pricelist_sale/models/sale_order.py (hunk 1 above)
Comment (nguyenminhchien) [reply account_invoice_pricelist_sale/models/sale_order.py]: updated

**Review by RodrigoBM — APPROVED** [review]

**Comment by RodrigoBM** [comment]
Can @hildickethan-S73  check it? Thanks

**Pull request merged**